policy/aggregation_type_options.go

```diff
@@ -0,0 +1,607 @@
+		P99,
+	}
+	defaultDefaultGaugeAggregationTypes = AggregationTypes{
+		Last,
+	}
+
+	defaultAggregationLastSuffix   = []byte(".last")
+	defaultAggregationSumSuffix    = []byte(".sum")
+	defaultAggregationSumSqSuffix  = []byte(".sum_sq")
+	defaultAggregationMeanSuffix   = []byte(".mean")
+	defaultAggregationMinSuffix    = []byte(".lower")
+	defaultAggregationMaxSuffix    = []byte(".upper")
+	defaultAggregationCountSuffix  = []byte(".count")
+	defaultAggregationStdevSuffix  = []byte(".stdev")
+	defaultAggregationMedianSuffix = []byte(".median")
+
+	defaultCounterSuffixOverride = map[AggregationType][]byte{
+		Sum: nil,
+	}
+	defaultTimerSuffixOverride = map[AggregationType][]byte{}
+	defaultGaugeSuffixOverride = map[AggregationType][]byte{
+		Last: nil,
+	}
+)
+
+type options struct {
+	defaultCounterAggregationTypes AggregationTypes
+	defaultTimerAggregationTypes   AggregationTypes
+	defaultGaugeAggregationTypes   AggregationTypes
+	sumSuffix                      []byte
+	sumSqSuffix                    []byte
+	meanSuffix                     []byte
+	lastSuffix                     []byte
+	minSuffix                      []byte
+	maxSuffix                      []byte
+	countSuffix                    []byte
+	stdevSuffix                    []byte
+	medianSuffix                   []byte
+	timerQuantileSuffixFn          QuantileSuffixFn
+	aggTypesPool                   AggregationTypesPool
+	quantilesPool                  pool.FloatsPool
+
+	defaultSuffixes [][]byte
+	counterSuffixes [][]byte
+	timerSuffixes   [][]byte
+	gaugeSuffixes   [][]byte
+
+	counterSuffixOverride map[AggregationType][]byte
+	timerSuffixOverride   map[AggregationType][]byte
+	gaugeSuffixOverride   map[AggregationType][]byte
+
+	defaultCounterAggregationSuffixes [][]byte
+	defaultTimerAggregationSuffixes   [][]byte
+	defaultGaugeAggregationSuffixes   [][]byte
+	timerQuantiles                    []float64
+}
+
+// NewAggregationTypesOptions returns a default AggregationTypesOptions.
+func NewAggregationTypesOptions() AggregationTypesOptions {
+	o := &options{
+		defaultCounterAggregationTypes: defaultDefaultCounterAggregationTypes,
+		defaultGaugeAggregationTypes:   defaultDefaultGaugeAggregationTypes,
+		defaultTimerAggregationTypes:   defaultDefaultTimerAggregationTypes,
+		lastSuffix:                     defaultAggregationLastSuffix,
+		minSuffix:                      defaultAggregationMinSuffix,
+		maxSuffix:                      defaultAggregationMaxSuffix,
+		meanSuffix:                     defaultAggregationMeanSuffix,
+		medianSuffix:                   defaultAggregationMedianSuffix,
+		countSuffix:                    defaultAggregationCountSuffix,
+		sumSuffix:                      defaultAggregationSumSuffix,
+		sumSqSuffix:                    defaultAggregationSumSqSuffix,
+		stdevSuffix:                    defaultAggregationStdevSuffix,
+		timerQuantileSuffixFn:          defaultTimerQuantileSuffixFn,
+		aggTypesPool:                   NewAggregationTypesPool(nil),
```

Comment:
nit: put the pool initialization code for the agg types pool and the quantiles pool into a `initPools` function for separation of concerns.